internal/timestamp/timestamp.go

```diff
@@ -52,5 +56,54 @@ func Timestamp(req *signature.SignRequest, opts tspclient.RequestOptions) ([]byt
 	if err := nx509.ValidateTimestampingCertChain(tsaCertChain); err != nil {
 		return nil, err
 	}
+	// certificate chain revocation check after timestamping
+	if req.TSARevocationValidator != nil {
+		certResults, err := req.TSARevocationValidator.ValidateContext(ctx, revocation.ValidateContextOptions{
+			CertChain: tsaCertChain,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("after timestamping: failed to check timestamping certificate chain revocation with error: %w", err)
+		}
+		if err := revocationFinalResult(certResults, tsaCertChain); err != nil {
+			return nil, fmt.Errorf("after timestamping: %w", err)
+		}
+	}
 	return resp.TimestampToken.FullBytes, nil
 }
+
+// revocationFinalResult returns an error if any cert in the cert chain has
+// a revocation status other than ResultOK or ResultNonRevokable.
+// When ResultRevoked presents, always return the revoked error.
+func revocationFinalResult(certResults []*result.CertRevocationResult, certChain []*x509.Certificate) error {
```

Comment:
Can we rename `revocationFinalResult` to some thing like `revocationResult` ? `Final` in the name doesn't improves readability.

Reply:
Updated.